From baa418b40408888f973f576a056d0ca8c824d703 Mon Sep 17 00:00:00 2001
From: Myles Watson <mylesgw@google.com>
Date: Sat, 27 Aug 2022 00:08:59 -0700
Subject: [PATCH 3/3] Controller: Warn on LeSetEventMask error codes

Bug: 239662211
Test: cert/run
Tag: #compatibility
Change-Id: Ib7769c0d7a2a3783d82cd9b561d8c38cca7b9c14
(cherry picked from commit b8b071e8186504ca4bd8e69fc69390f361ec96bc)
---
 system/gd/hci/controller.cc | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/system/gd/hci/controller.cc b/system/gd/hci/controller.cc
index ad00ecb767..50c8182542 100644
--- a/system/gd/hci/controller.cc
+++ b/system/gd/hci/controller.cc
@@ -541,8 +541,18 @@ struct Controller::impl {
 
   void le_set_event_mask(uint64_t le_event_mask) {
     std::unique_ptr<LeSetEventMaskBuilder> packet = LeSetEventMaskBuilder::Create(le_event_mask);
-    hci_->EnqueueCommand(std::move(packet), module_.GetHandler()->BindOnceOn(
-                                                this, &Controller::impl::check_status<LeSetEventMaskCompleteView>));
+    hci_->EnqueueCommand(
+        std::move(packet), module_.GetHandler()->BindOnceOn(this, &Controller::impl::check_le_set_event_mask_status));
+  }
+
+  void check_le_set_event_mask_status(CommandCompleteView view) {
+    ASSERT(view.IsValid());
+    auto status_view = LeSetEventMaskCompleteView::Create(view);
+    ASSERT(status_view.IsValid());
+    auto status = status_view.GetStatus();
+    if (status != ErrorCode::SUCCESS) {
+      LOG_WARN("Unexpected return status %s", ErrorCodeText(status).c_str());
+    }
   }
 
   template <class T>
-- 
2.24.3 (Apple Git-128)

